spec/rubocop/cop/style/if_with_boolean_literal_branches_spec.rb

```diff
@@ -0,0 +1,419 @@
+      expect_no_offenses(<<~RUBY)
+        unless condition
+          false
+        else
+          true
+        end
+      RUBY
+    end
+
+    it 'does not register an offense when using `if condition` with opposite boolean literal branches' do
+      expect_no_offenses(<<~RUBY)
+        if condition
+          false
+        else
+          true
+        end
+      RUBY
+    end
+
+    it 'does not register an offense when using `unless condition` with opposite boolean literal branches' do
+      expect_no_offenses(<<~RUBY)
+        unless condition
+          true
+        else
+          false
+        end
+      RUBY
+    end
+  end
+
+  context 'when condition is a logical operator and operands do not known whether to return boolean value' do
+    it 'does not register an offense when using `if foo && bar` with boolean literal branches' do
+      expect_no_offenses(<<~RUBY)
+        if foo && bar
+          true
+        else
+          false
+        end
+      RUBY
+    end
+
+    it 'does not register an offense when using `if foo || bar` with boolean literal branches' do
+      expect_no_offenses(<<~RUBY)
+        if foo || bar
+          true
+        else
+          false
+        end
+      RUBY
+    end
+
+    it 'does not register an offense when using `unless foo && bar` with boolean literal branches' do
+      expect_no_offenses(<<~RUBY)
+        unless foo && bar
+          false
+        else
+          true
+        end
+      RUBY
+    end
+
+    it 'does not register an offense when using `unless foo || bar` with boolean literal branches' do
+      expect_no_offenses(<<~RUBY)
+        unless foo || bar
+          false
+        else
+          true
+        end
+      RUBY
+    end
+
+    it 'does not register an offense when using `if foo && bar` with opposite boolean literal branches' do
+      expect_no_offenses(<<~RUBY)
+        if foo && bar
+          false
+        else
+          true
+        end
+      RUBY
+    end
+
+    it 'does not register an offense when using `if foo || bar` with opposite boolean literal branches' do
+      expect_no_offenses(<<~RUBY)
+        if foo || bar
+          false
+        else
+          true
+        end
+      RUBY
+    end
+
+    it 'does not register an offense when using `unless foo && bar` with opposite boolean literal branches' do
+      expect_no_offenses(<<~RUBY)
+        unless foo && bar
+          true
+        else
+          false
+        end
+      RUBY
+    end
+
+    it 'does not register an offense when using `unless foo || bar` with opposite boolean literal branches' do
+      expect_no_offenses(<<~RUBY)
+        unless foo || bar
+          true
+        else
+          false
+        end
+      RUBY
+    end
+
+    it 'does not register an offense when using `if foo? && bar && baz?` with boolean literal branches' do
+      expect_no_offenses(<<~RUBY)
+        if foo? && bar && baz?
```

Comment:
Why no offense here? I understand not wanting to inverse a complex condition, but to me anything not opposite is better simplified, no?

Reply:
It's my mistake. 💦 I refined the conditions of `&&` and `||` again. Updated implementation registers an offense if it can be assumed that a boolean value will be returned.